### PR TITLE
Data synchronization retry logic optimization

### PR DESCRIPTION
**Pull request by pengzhengfa**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by pengzhengfa** [comment]
@KomachiSion  Hello, please approve

**Comment by KomachiSion** [comment]
Can you create a issue to describe why you do these change first?

**Comment by pengzhengfa** [comment]
@KomachiSion ok

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/TaskDispatcher.java [hunk 1]
```diff
@@ -65,7 +67,7 @@ public void addTask(String key) {
 
         private int index;
 
-        private int dataSize = 0;
+        private AtomicInteger dataSize = new AtomicInteger(0);
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/TaskDispatcher.java:70]: There is no need, the task dispatcher is a singleton.

File: naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/DataSyncer.java [hunk 2]
```diff
@@ -50,12 +52,14 @@
     private final Serializer serializer;
     private final DistroMapper distroMapper;
     private final ServerMemberManager memberManager;
-
+    private final int firstRetryCount = 1;
+    private final int maxRetryCount = 10;
+    private AtomicInteger requestCount = new AtomicInteger(0);
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/DataSyncer.java:57]: two problem:
1. requestCount will be shared between different task. Task A failed, Task B failed, the request count will be 2, but one of them should only wait 1 multiple delay.
2. no reset to 0 for requestCount.

**Comment by KomachiSion** [comment]
Close reason see #3081 .

**Pull request closed without merging**